core/src/main/scala/cats/syntax/list.scala

```diff
@@ -51,6 +49,80 @@ final class ListOps[A](private val la: List[A]) extends AnyVal {
     toNel.fold(SortedMap.empty[B, NonEmptyList[A]])(_.groupBy(f))
   }
 
+  /**
+   * Groups elements inside this `List` according to the `Order` of the keys
+   * produced by the given key function.
+   * And each element in a group is transformed into a value of type B
+   * using the mapping function.
+   *
+   * {{{
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.data.NonEmptyList
+   * scala> import cats.implicits._
+   * scala> val list = List(12, -2, 3, -5)
+   * scala> val expectedResult = SortedMap(false -> NonEmptyList.of("-2", "-5"), true -> NonEmptyList.of("12", "3"))
+   * scala> list.groupMapNel(_ >= 0)(_.toString) === expectedResult
+   * res0: Boolean = true
+   * }}}
+   */
+  def groupMapNel[K, B](key: A => K)(f: A => B)(implicit K: Order[K]): SortedMap[K, NonEmptyList[B]] = {
+    implicit val ordering: Ordering[K] = K.toOrdering
+    toNel.fold(SortedMap.empty[K, NonEmptyList[B]])(_.groupMap(key)(f))
+  }
+
+  /**
+   * Groups elements inside this `List` according to the `Order` of the keys
+   * produced by the given key function.
+   * Then each element in a group is transformed into a value of type B
+   * using the mapping function.
+   * And finally they are all reduced into a single value
+   * using their `Semigroup`.
+   *
+   * {{{
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.implicits._
+   * scala> val list = List("Hello", "World", "Goodbye", "World")
+   * scala> val expectedResult = SortedMap("goodbye" -> 1, "hello" -> 1, "world" -> 2)
+   * scala> list.groupMapReduce(_.trim.toLowerCase)(_ => 1) === expectedResult
+   * res0: Boolean = true
+   * }}}
+   */
+  def groupMapReduce[K, B](key: A => K)(f: A => B)(implicit K: Order[K], B: Semigroup[B]): SortedMap[K, B] =
```

Comment:
Not sure how to name this one so it doesn't clash with the one introduced by the stdlib in `2.13`

Also, not sure if it worth adding since in `2.13` one can easily pass the `combine` operation provided by the **Semigroup**.